### DISCUSSION TITLE
Pupillary offline measurements with PupilEXT

### DISCUSSION THREAD

**Discussion by royquesada** [openPupil/Open-PupilEXT#23] [General]
Hello
I wonder if someone can tell me the steps to perform the offline processing starting from already acquired images without using PupilEXT or calibration files (and get the pupil diameter in pixels). I have watched the author's 3 videos but this case is not covered in them.
I am currently doing the following:
File/Open Images Directory and then play directory but it does not show me the pupillary diameter in any way. 

Thanks to anyone who can be of help

I am attaching some photo of what is happening to me 
![1-pupilext](https://user-images.githubusercontent.com/100945851/156776762-a08859bb-4d8c-47f9-9b16-a43289ce240c.JPG)
![PupilEXT_offline mode](https://user-images.githubusercontent.com/100945851/156776792-20632d20-aaa8-4623-a862-8f05b1641c53.png)

**Comment by BZandi** [comment]
PupilEXT can only track one eye, so you need to adjust the region of interest (ROI) in the live view panel. For this, you need to activate the option "Use ROI Area Selection" in the pupil detection settings panel. Then, go to the live image view, press "ROI", and select an ROI ratio (for example, 0.6). Then, place the appearing ROI area over one eye (zoom into the image if necessary to be more precise with the placement). You can also adjust the size of the ROI. The pupil detection will then only iterate in the selected ROI area. After setting the ROI, you can start with tracking the pupil size (see video below):


https://user-images.githubusercontent.com/62992112/156878057-16d7d8e7-473a-4ed2-b0ad-df958b9bf9c5.mp4



**Important**
It may be that the quality of the pupil tracking is not good for your setting, which is because the parameters of the algorithm are not optimally adjusted to your adjusted ROI size or image resolution. It is recommended to use the PuRe algorithm. Then open the "Pupil Detection Settings" window. There you have to adjust the algorithm's parameters to the selected ROI size. You currently adjusted the PuRe algorithm with its default parameter as provided in your screenshot. We provide the presets for the ROI sizes 0.6 and 0.3. Click on the selection bar and change it from "Default" to one of the provided preset to enhance the pupil tracking quality. You can also manually adjust the algorithm's parameters to improve the accuracy of the pupil tacking. It should be straightforward. Do not forget to press apply in the "Pupil Detection Settings" window, otherwise the adjustments will not be active.

Note that we also provide a walkthrough on offline analysis in the third video of our tutorials. The data we use in this video can be downloaded so you can use the provided images to play around with PupilEXT:

Tutorial offline analysis in PupilEXT:
[https://vimeo.com/564778587](https://vimeo.com/564778587)

Recorded eye images that can be downloaded to play around with PupilEXT:
[https://tudatalib.ulb.tu-darmstadt.de/handle/tudatalib/2929](https://tudatalib.ulb.tu-darmstadt.de/handle/tudatalib/2929)

**Edit:**
When loading images to PupilEXT for offline analysis, take care of proper file names. For example, if you label your files with sequenced numbers such as 1.png, 2.png, 3.png, 4.png, 5.png, 6.png, 7.png, 8.png, 9.png, 10.png, 11.png, you will run into issues, as the order of the images will be changed to 1.png, 10.png, 11.png, 2.png, ...
It is better to label the images with 001.png, with the time stamp in milliseconds or similar. If you recorded the images with PupilEXT, everything is labelled correctly as we use the timestamp in milliseconds as filenames. The provided demo datasets are recorded with PupilEXT.

**Reply by royquesada** [reply]
Thank you! Your answer helped me a lot
